### PR TITLE
[CARBONDATA-2982] CarbonSchemaReader support array<string>

### PR DESCRIPTION
**Pull request by xubo245**
This PR fix the issue and change :
org.apache.carbondata.sdk.file.CarbonSchemaReader#readSchemaInDataFile
org.apache.carbondata.sdk.file.CarbonSchemaReader#readSchemaInIndexFile

This PR remove child schema

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
     add test code
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/620/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8881/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/812/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/623/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8884/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/815/

File: store/sdk/src/test/java/org/apache/carbondata/sdk/file/CarbonReaderTest.java [hunk 1]
```diff
@@ -1435,5 +1435,99 @@ public void testReadWithFilterOfnonTransactionalwithsubfolders() throws IOExcept
...
+            row[8], row[9]
+        ));
+        Object[] arr = (Object[]) row[10];
+        for (int j = 0; j < arr.length; j++) {
+          System.out.print(arr[j] + " ");
+        }
+        System.out.println();
+        i++;
+      }
+      System.out.println("\nFinished");
```
Comment (KanakaKumar) [inline store/sdk/src/test/java/org/apache/carbondata/sdk/file/CarbonReaderTest.java]: Please remove Sys out content

**Comment by ajantha-bhat** [comment]
LGTM except for minor test case comment by @KanakaKumar .

File: store/sdk/src/test/java/org/apache/carbondata/sdk/file/CarbonReaderTest.java (hunk 1 above)
Comment (xubo245) [reply store/sdk/src/test/java/org/apache/carbondata/sdk/file/CarbonReaderTest.java]: ok, done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/842/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8910/

**Comment by xubo245** [comment]
@KanakaKumar @ajantha-bhat removed

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8914/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/846/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/850/

**Comment by manishgupta88** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/679/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/874/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8942/

**Pull request closed without merging**